src/github.com/matrix-org/dendrite/roomserver/alias/alias.go

```diff
@@ -98,14 +98,30 @@ func (r *RoomserverAliasAPI) GetAliasRoomID(
 	return nil
 }
 
+// GetAliasesForRoomID implements api.RoomserverAliasAPI
+func (r *RoomserverAliasAPI) GetAliasesForRoomID(
+	ctx context.Context,
+	request *api.GetAliasesForRoomIDRequest,
+	response *api.GetAliasesForRoomIDResponse,
+) error {
+	// Look up the aliases in the database for the given RoomID
+	aliases, err := r.DB.GetAliasesForRoomID(ctx, request.RoomID)
+	if err != nil {
+		return err
+	}
+
+	response.Aliases = aliases
+	return nil
+}
```

Comment:
Where was the implementation for `GetAliasesFromRoomID`? I'm surprised I'm not seeing that removed in the diff.

Reply:
There wasn't one. I've had to add it :)

Reply:
Then how was `RoomserverAliasAPIDatabase` implemented?

Reply:
The database queries and supporting functions were already there, a method to expose it to the RoomserverAliasAPI was all that was needed.